proposals/nnnn-screen-manager-audio-pass-thru.md

```diff
@@ -0,0 +1,120 @@
+    private boolean muteAudio;
+    private Integer timeout;
+    private SDLAudioPassThruCapabilities capabilities;
+    
+    // All vars have getters / setters
+
+    public MicrophoneInputView(@Nullable String text, @Nullable String secondaryText, @Nullable PlayAudioData audioPrompt, @Nullable boolean muteAudio, @Nullable Integer timeout, @Nullable capabilities)
+}
+```
+
+Note that the `PlayAudioData` class comes from another proposal that has not been accepted as of the writing of this proposal, and is [found here](https://github.com/smartdevicelink/sdl_evolution/pull/928/files#diff-05979b405babd4a720b6d0f3ecb98e9dR22).
+
+### Screen Manager Updates
+##### iOS
+```objc
+@interface SDLScreenManager
+// Existing properties and methods
+
+
+/// Called when new audio data arrives.
+typedef void (^SDLMicrophoneDataHandler)(NSData *__nullable audioData);
+
+/// Create a view to receive raw microphone input information from the head unit. This maps to the SDLAudioPassThru RPC.
+/// @param microphoneInputView The view that will be presented to the screen and various capabilities related to the microphone information that will be retrieved.
+/// @param microphoneDataHandler A handler that will be called with raw audio data
+/// @completionHandler A handler sent when the head unit sends a response and a possible error if one occurs.
+- (void)presentMicrophoneInputView:(SDLMicrophoneInputView *)microphoneInputView withAudioDataHandler:(SDLMicrophoneDataHandler)microphoneDataHandler completionHandler:(nullable SDLScreenManagerUpdateHandler)completionHandler;
+
+/// Dismisses the current microphone input view if there is one. If you're currently receiving audio data, you may not immediately cease to receive this data.
+/// @param completionHandler A possible error if one occurs
+- (void)dismissMicrophoneInputViewWithCompletionHandler:(nullable SDLScreenManagerUpdateHandler)completionHandler;
+
+@end
+```
+
+##### Java
+```java
+public interface AudioDataListener {
+    void onAudioData(byte[] audioData)
+}
+
+public class BaseScreenManager {
+    public void presentMicrophoneInputView(@NonNull MicrophoneInputView view, @NonNull AudioDataListener audioDataListener, @Nullable CompletionListener completionListener)
+    public void dismissMicrophoneInputView(@Nullable CompletionListener completionListener)
+}
+```
+
+### JavaScript Suite APIs
+Due to the size of the iOS APIs and the similarity between the iOS, Java Suite and eventual JavaScript Suite APIs, this proposal does not present the public APIs of the JavaScript Suite APIs – especially because the JavaScript Suite APIs do not currently have a screen manager layer. The JavaScript Suite APIs should mirror the iOS and Java Suite API appropriately and is up to the maintainers' discretion. However, if any changes needed to be made such that they impacted the iOS / Java Suite API (such as the alteration, addition, or removal of a method or property), then a proposal revision would be needed.
```

Comment:
```suggestion
Due to the size of the iOS APIs and the similarity between the iOS, Java Suite and eventual JavaScript Suite APIs, this proposal does not present the public APIs of the JavaScript Suite APIs – especially because the JavaScript Suite APIs do not currently have a screen manager layer. The JavaScript Suite APIs should mirror the iOS and Java Suite API appropriately and is up to the Project Maintainer's discretion. However, if any changes needed to be made such that they impacted the iOS / Java Suite API (such as the alteration, addition, or removal of a method or property), then a proposal revision would be needed.
```
Please change "maintainers'" to be more specific, "Project Maintainer's"